pyte/screens.py

```diff
@@ -592,11 +605,13 @@ def insert_lines(self, count=None):
 
         # If cursor is outside scrolling margins it -- do nothin'.
         if top <= self.cursor.y <= bottom:
-            #                           v +1, because range() is exclusive.
-            for line in range(self.cursor.y,
-                              min(bottom + 1, self.cursor.y + count)):
-                self.buffer.pop(bottom)
-                self.buffer.insert(line, take(self.columns, self.default_line))
+            # Shift old lines down (backwards)
+            for old_line in range(bottom, self.cursor.y - 1, -1):
+                if old_line + count <= bottom:  # There is place inside margins
+                    self.buffer[old_line + count] = self.buffer[old_line]
```

Comment:
I'll play with it when I will work on performance improving. 

Now it just create overhead on checking if `old_line+count` and `old_line` exists in buffer, because them always exists and have minimal cost of moving.